pyquil/api/_qpu.py

```diff
@@ -292,3 +293,18 @@ def _resolve_memory_references(self, expression: Expression) -> Union[float, int
             return self._variables_shim.get(ParameterAref(name=expression.name, index=expression.offset), 0)
         else:
             raise ValueError(f"Unexpected expression in gate parameter: {expression}")
+
+    @_record_call
+    def reset(self):
+        """
+        Reset the state of the underlying QAM, and the QPU Client connection.
+        """
+        super().reset()
+
+        def refresh_client(client: Client, new_endpoint: str) -> Client:
```

Comment:
Can this be deleted, since it duplicates the top-level `reset`?

Reply:
Can you explain what you mean? I don't see how a QPU's client is refreshed any place but here?

Reply:
https://github.com/rigetti/pyquil/pull/872/files#r317032414

Reply:
The function is duplicated.

Reply:
yes its in two spots ur right

Reply:
where should i put it? i wasnt sure. base_connection?